### DISCUSSION TITLE
Resource Allocation in Artery

### DISCUSSION THREAD

**Discussion by matheuspleal** [riebl/artery#293] [Q&A] [answered]
I need to implement a computational resource allocation algorithm for my master's thesis. I've seen some work using Veins, but I need to use a V2X environment, so I chose Artery. Is there any work or any tips for me to be able to implement what I need?

**Comment by awillecke** [comment, marked as answer]
Hi,

I'm not aware of any works regarding allocation of computational resources for Artery. AFAIK, Veins and also Artery do not model computation time and memory consumption of software, running on each vehicle instance. They just assume that memory resources are sufficient and calculations can be completed. Usually, all computations, e.g. execution of a modules functions, are performed for an [event](https://doc.omnetpp.org/omnetpp/manual/#sec:simple-modules:discrete-event-simulation) and have no duration in the simulation world.

If you want to model computational resources for service, a good starting point would be the [middleware](http://artery.v2x-research.eu/features/middleware/) since it is responsible for running services on each equipped [vehicle](https://github.com/riebl/artery/blob/4bc468e372091fa392d43c5fc3358821a21534c9/src/artery/inet/Car.ned#L74).

Best regards
Alex

**Reply by matheuspleal** [reply]
Excellent! Thanks for the answer.

I'm going to study about Middleware to develop the structure that I need then.
